File: test/Exception1.java
import java.util.*;

public class Exception1 {
    int getSum(int arr[], int choice[]) {
        int i, total = 0;
        for (i = 0; i < choice.length; i++) {
            total += arr[choice[i]];
        }

        return total;
    }

    double getSqrtSum(int arr[], int choice[]) {
        int i;
        double total = 0.0;
        for (i = 0; i < choice.length; i++) {
            total += Math.sqrt(arr[choice[i]]);
        }

        return total;
    }

    void display(int arr[]) {
        int i;
        for (i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Exception1 ob = new Exception1();
        int i, n, ch;

        System.out.println("Enter size");
        n = sc.nextInt();
        int arr[] = new int[n];
        int choice[] = new int[5];

        System.out.println("Enter elements:");
        for (i = 0; i < n;) {
            try {
                ch = sc.nextInt();
            } catch (InputMismatchException e) {
                System.out.println("Invalid Input. Enter again");
                sc.next();
                continue;
            }
            arr[i++] = ch;
        }

        System.out.println("Enter 5 indices:");
        for (i = 0; i < 5;) {
            try {
                ch = sc.nextInt();
                if (ch < 0 || ch >= n) {
                    throw new Exception("Invalid Index");
                }
            } catch (InputMismatchException e) {
                System.out.println("Invalid Input. Enter again");
                sc.next();
                continue;
            } catch (Exception e) {
                System.out.println(e.getMessage());
                // sc.next();
                continue;
            }

            choice[i++] = ch;
        }

        System.out.print("Arr = ");
        ob.display(arr);
        System.out.print("Choice = ");
        ob.display(choice);

        int res1;
        double res2;
        res1 = ob.getSum(arr, choice);
        res2 = ob.getSqrtSum(arr, choice);

        System.out.println("Sum = " + res1);
        System.out.println("Sqrt Sum = " + res2);

        sc.close();
    }
}

File: test/Exception1.cpp
#include <iostream>
#include <cmath>
#include <limits>
#include <vector>
using namespace std;

class Extension1
{
public:
    int getSum(const vector<int> &arr, const vector<int> &choice)
    {
        int i, total = 0;
        for (i = 0; i < choice.size(); i++)
        {
            total += arr[choice[i]];
        }

        return total;
    }

    double getSqrtSum(const vector<int> &arr, const vector<int> &choice)
    {
        int i;
        double total = 0.0;
        for (i = 0; i < choice.size(); i++)
        {
            total += sqrt(arr[choice[i]]);
        }

        return total;
    }

    void display(const vector<int> &arr)
    {
        int i;
        for (i = 0; i < arr.size(); i++)
        {
            cout << arr[i] << " ";
        }
        cout << endl;
    }
};

int main()
{
    Extension1 ob;
    int n, ch, i;
    cout << "Enter no. of inputs: ";
    cin >> n;

    vector<int> arr(n);
    vector<int> choice(5);

    cout << "Enter the elements:" << endl;
    for (i = 0; i < n;)
    {
        if (!(cin >> ch))
        {
            cout << "Invalid input. Enter again" << endl;
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
            continue;
        }
        arr[i++] = ch;
    }

    cout << "Choose 5 indices:" << endl;
    for (i = 0; i < 5;)
    {
        if (!(cin >> ch) || ch < 0 || ch >= n)
        {
            cout << "Invalid input. Enter again" << endl;
            cin.clear();
            cin.ignore(numeric_limits<streamsize>::max(), '\n');
            continue;
        }
        choice[i++] = ch;
    }

    cout << "Inputs: ";
    ob.display(arr);
    cout << "Choices: ";
    ob.display(choice);

    int sum;
    double res;
    sum = ob.getSum(arr, choice);
    res = ob.getSqrtSum(arr, choice);

    cout << "Sum = " << sum << endl;
    cout << "Sqrt Sum = " << res << endl;

    return 0;
}